### PR TITLE
Include `Custom-AI-rules`; docs touchups

### PR DESCRIPTION
**Pull request by Haleshot**
## 📝 Summary

Add `Custom AI rules` in [AI completion docs](https://docs.marimo.io/guides/editor_features/ai_completion.html#ai-completion) along with relevant image for reference; other minor doc changes.

## 🔍 Description of Changes

- Hyperlinking to relevant doc webpages for easier navigation
- Rectify position of `allow_attachments` parameter in function paramter
- Describe `marimo.toml` config a little (also hyperlinking to it's relevant webpage)
- Touches on #2773 (should update [`chat.md`](https://docs.marimo.io/api/inputs/chat.html#chat) to for it to completely address the issue), https://github.com/marimo-team/marimo/pull/2493, 

## 📋 Checklist

- [ ] I have read the [contributor guidelines](https://github.com/marimo-team/marimo/blob/main/CONTRIBUTING.md).
- [ ] For large changes, or changes that affect the public API: this change was discussed or approved through an issue, on [Discord](https://discord.gg/JE7nhX6mD8), or the community [discussions](https://github.com/marimo-team/marimo/discussions) (Please provide a link if applicable).
- [ ] I have added tests for the changes made.
- [ ] I have run the code and verified that it works as expected.

## 📜 Reviewers

@akshayka OR @mscolnick

### CODE REVIEW

File: docs/guides/editor_features/ai_completion.md [hunk 1]
```diff
@@ -71,14 +77,51 @@ activate_on_typing = true
...
-the bottom of your notebook. You can also refactor existing cells by inputting
-`Ctrl/Cmd-Shift-e` in a cell, opening an input to modify the cell using AI.
+You can customize how the AI assistant behaves by adding rules in the marimo settings. These rules help ensure consistent code generation across all AI providers. You can find more information about marimo's supported plotting libraries and data handling in the [Plotting Guide](https://docs.marimo.io/guides/working_with_data/plotting.html#plotting) and [Data Working Guide](https://docs.marimo.io/guides/working_with_data/index.html).
+
+<div align="center">
+  <figure>
+    <img src="/_static/docs-ai-completion-custom-assist-rules.png"/>
+    <figcaption>Configure custom AI rules in settings</figcaption>
+  </figure>
+</div>
```
Comment (Haleshot) [inline docs/guides/editor_features/ai_completion.md:91]:
```suggestion
You can customize how the AI assistant behaves by adding rules in the marimo settings. These rules help ensure consistent code generation across all AI providers. You can find more information about marimo's supported plotting libraries and data handling in the [Plotting Guide](https://docs.marimo.io/guides/working_with_data/plotting.html#plotting) and [Data Working Guide](https://docs.marimo.io/guides/working_with_data/index.html).

<div align="center">
  <figure>
    <img src="/_static/docs-ai-completion-custom-assist-rules.png"/>
    <figcaption>Configure custom AI rules in settings</figcaption>
  </figure>
</div>

Should I follow https://docs.marimo.io/guides/configuration/index.html and align the image to the right with text on the left?
```

File: docs/guides/editor_features/ai_completion.md [hunk 2]
```diff
@@ -162,4 +182,48 @@ model = "gemini-1.5-flash"
...
+base_url = "http://127.0.0.1:11434/v1"
+```
+
+### Using other AI providers
+
+marimo supports OpenAI's API by default. Many providers offer OpenAI API-compatible endpoints, which can be used by simply changing the `base_url` in your configuration. For example, providers like [GROQ](https://console.groq.com/docs/openai) follow this pattern.
+
+For a comprehensive list of compatible providers and their configurations, please refer to the [liteLLM Providers documentation](https://litellm.vercel.app/docs/providers).
+
+For providers not compatible with OpenAI's API, please submit a [feature request](https://github.com/marimo-team/marimo/issues/new?template=feature_request.yaml) or "thumbs up" an existing one.
```
Comment (mscolnick) [inline docs/guides/editor_features/ai_completion.md:229]: nice this is great!

File: docs/guides/editor_features/ai_completion.md (hunk 1 above)
Comment (mscolnick) [reply docs/guides/editor_features/ai_completion.md:91]: i think centered is great

**Review by mscolnick — DISMISSED** [review]
looks great!

File: docs/guides/editor_features/ai_completion.md [hunk 3]
```diff
@@ -16,8 +16,14 @@ _Get started with Copilot_:
...
-_Note_: Copilot is not yet available in our conda distribution; please install
-marimo using `pip` if you need Copilot.
+```{admonition} Built from the ground up
+:class: note
+   marimo comes with GitHub Copilot, a tool that helps you write code faster by
+   suggesting in-line code suggestions based on the context of your current code.
+
+   marimo also comes with the ability to use AI for refactoring a cell, finishing writing a cell, or writing a full cell from scratch.
+   This feature is currently experimental and is not enabled by default.
+```
```
Comment (akshayka) [inline docs/guides/editor_features/ai_completion.md:26]: @Haleshot , this appears to be duplicated at the top of the page.

Additionally, the admonition title seems to have been copied from a different title.

**Review by akshayka — COMMENTED** [review]
Just one comment, looks great!

File: docs/guides/editor_features/ai_completion.md [hunk 4]
```diff
@@ -71,14 +77,51 @@ activate_on_typing = true
...
-OpenAI and Anthropic, as well as local models served via Ollama.
+marimo has built-in support for generating and refactoring code with AI, with a variety of providers. marimo works with hosted AI providers, such as OpenAI, Anthropic, and Google, as well as local models served via Ollama.
+
+### Custom AI Rules
 
-Below we describe how to connect marimo to your AI provider. Once enabled, you
-can generate entirely new cells by clicking the "Generate with AI" button at
-the bottom of your notebook. You can also refactor existing cells by inputting
-`Ctrl/Cmd-Shift-e` in a cell, opening an input to modify the cell using AI.
+You can customize how the AI assistant behaves by adding rules in the marimo settings. These rules help ensure consistent code generation across all AI providers. You can find more information about marimo's supported plotting libraries and data handling in the [Plotting Guide](https://docs.marimo.io/guides/working_with_data/plotting.html#plotting) and [Data Working Guide](https://docs.marimo.io/guides/working_with_data/index.html).
```
Comment (akshayka) [inline docs/guides/editor_features/ai_completion.md:84]:
```suggestion
You can customize how the AI assistant behaves by adding rules in the marimo settings. These rules help ensure consistent code generation across all AI providers. You can find more information about marimo's supported plotting libraries and data handling in the [plotting guide](https://docs.marimo.io/guides/working_with_data/plotting.html#plotting) and [working with data guide](https://docs.marimo.io/guides/working_with_data/index.html).
```

File: docs/guides/configuration/index.md [hunk 5]
```diff
@@ -45,9 +45,9 @@ You can customize the following:
 - Completion (auto-completion, AI copilot, etc.)
 - Display (theme, font size, output placement, etc.)
 - Autosave 
-- Package management 
+- [Package management](https://docs.marimo.io/guides/editor_features/package_management.html#package-management) 
 - Server settings
-- VIM keybindings
+- [VIM keybindings](https://docs.marimo.io/guides/editor_features/overview.html#vim-keybindings)
```
Comment (mscolnick) [inline docs/guides/configuration/index.md:50]: i think you can hyperlink by file url, then we may get errors if the file changes

Comment (mscolnick) [reply docs/guides/configuration/index.md:50]: not sure if anchor's work though, could try:
`/guides/editor_features/overview#vim-keybindings`

File: docs/guides/configuration/index.md [hunk 6]
```diff
@@ -21,7 +21,7 @@ App Configuration is specific to each notebook and is stored in the `notebook.py
 
 - Notebook width
 - Notebook title
-- Custom CSS
+- [Custom CSS](https://docs.marimo.io/guides/configuration/theming.html)
```
Comment (mscolnick) [inline docs/guides/configuration/index.md:24]: i think `/guides/configuration/theming` should just work

**Review by mscolnick — APPROVED** [review]
Thank you!

**Pull request merged**